app/templates/profile.html

```diff
@@ -1,285 +1,296 @@
+                                            <button type="submit" class="btn btn-sm btn-success">Save changes</button>
+                                        </div>
+                                    </form>
+                                </div>
+
                             </div>
                         </div>
                     </div>
-                    <!-- End profile update menu -->
+                </div>
+                <!-- End profile update menu -->
 
-                    <img class="card-img-top profile-image rounded-circle mx-auto d-block"
-                     src="{{ url_for('media', path=user.avatar) }}" alt="Profile image">
-                    <div class="card-body">
-                        <h6 class="card-title text-center mb-1">{{ user.full_name }}</h6>
-                        <p class="card-text text-center mt-0">
-                            <a class="nav-link text-secondary" href="#">
-                                <i class="fa fa-spin fa-cog" aria-hidden="true"></i>
-                                Settings
-                            </a>
-                        </p>
-                        <hr>
-                        <p class="card-text text-center">
-                            <small>
+                <img class="card-img-top profile-image rounded-circle mx-auto d-block" src="{{ url_for('media', path=user.avatar) }}" alt="Profile image">
+                <div class="card-body">
+                    <h6 class="card-title text-center mb-1">{{ user.full_name }}</h6>
+                    <p class="card-text text-center mt-0">
+                        <a class="nav-link text-secondary" href="#">
+                            <i class="fa fa-spin fa-cog" aria-hidden="true"></i> Settings
+                        </a>
+                    </p>
+                    <hr>
+                    <p class="card-text text-center">
+                        <small>
                                 {{ user.description }}
                             </small>
-                        </p>
-                    </div>
+                    </p>
                 </div>
-                <!-- End Profile card -->
+            </div>
+            <!-- End Profile card -->
+            <script src="../static/horoscope.js"></script>
+
+            <div class="card card-profile">
+                <div class="card-body">
+                    <p class="card-title">
+                        <div class="btn-group">
+                            <button type="button" class="btn btn-secondary dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+                                daily horoscope
+                            </button>
+                            <div class="dropdown-menu dropdown-menu-right">
+                                <button onclick="daily_horoscope('aries')" class="dropdown-item" type="button">Aries</button>
```

Comment:
Use Python loop, and please use JavaScript's addEventListener instead of using `onclick=` inside the HTML :)

Reply:
is it important to change to addEventListener ?
it make it nearly impossible
onclick works perfect here, i can't make it work with addEventListener

Reply:
I will explain the problem:
I add eventlistener to the drop-down menu to each element, after I did it doesn't work anymore.
can't make it work.
i fixed all the other problems.

Reply:
Add the Listener to each button using JavaScript. Be sure that it runs only after the document has been loaded.

Reply:
I tried that yesterday for 7 hours and didn't find a way to make it work.
I will try again today but i don't understand why it is better then onclick

Reply:
Done, I made it work